packages/warriorjs-abilities/src/look.js

```diff
@@ -7,7 +7,14 @@ function look({ range }) {
     description: `Returns an array of up to ${range} spaces in the given direction (${defaultDirection} by default).`,
     perform(direction = defaultDirection) {
       const offsets = Array.from(new Array(range), (_, index) => index + 1);
-      return offsets.map(offset => unit.getSensedSpaceAt(direction, offset));
+      const sensedSpaces = [];
+      offsets
+        .map(offset => unit.getSensedSpaceAt(direction, offset))
+        .some(space => {
```

Comment:
Have you thought about using `spaces.slice()` instead of `.some()` and an auxiliary array? I think it's cleaner than `.some()`.

`firstWallIndex` can be calculated with `spaces.findIndex()`. If a wall is found, then slice the spaces array up to that wall. If not, return all spaces. What do you think?